### ISSUE TITLE
PANIC: unprotected error in call to Lua API (bad argument #3 to '?' (function expected, got nil))

### ISSUE THREAD

**Issue by nyngwang** [neovim/neovim#16688]
### Neovim version (nvim -v)

0.7

### Vim (not Nvim) behaves the same?

not related

### Operating system/version

macOS 11.6.2

### Terminal name/version

alacritty 0.9.0 (fed349a)

### $TERM environment variable

alacritty

### Installation

```
brew install neovim --HEAD
```

### How to reproduce the issue

I was reported that this line of my `init.lua` is troublesome:

```
vim.g.mapleader = ' '
```

I was editing the `init.lua` when I was running `brew upgrade --fetch-HEAD` to update neovim. I suspect that this is related. But I have tried rebooting/reinstalling neovim, but all result in the same error.

### Expected behavior

no error

### Actual behavior

```
rror detected while processing /Users/foo/.config/nvim/init.lua:
E5113: Error while calling lua chunk: /Users/foo/.config/nvim/init.lua:7: attempt to index field 'g' (a nil value)
stack traceback:
        /Users/foo/.config/nvim/init.lua:7: in main chunk
Press ENTER or type command to continuePANIC: unprotected error in call to Lua API (bad argument #3 to '?' (function expected, got nil))
                                                                                                                                        %      
```

Now I go back to `brew install neovim` and there is no error.

**Comment by kuntau** [comment]
Running `nvim -u NONE` also give the same error 😭

**Comment by nyngwang** [comment]
@kuntau Now I go back to brew install neovim and there is no error. I think this is a temporary solution.

**Comment by clason** [comment]
The error indicates that you did not correctly _install_ Neovim, since the "standard library" is missing. I suspect mixing runtime and binary from different versions. Remove all traces of Neovim from your system and reinstall (`brew install --HEAD neovim` if you _must_ use a prerelease version, which I wouldn't recommend to you).

**Comment by nyngwang** [comment]
I have removed both `.local/` and `.config/nvim/` folders and reboot my computer. How could I know where to delete those traces?

**Comment by clason** [comment]
I mean the Neovim installation (wherever that is, e.g., `/usr/local/bin/nvim` and `/usr/local/share/nvim`).

**Comment by clason** [comment]
I'm closing because it's not reproducible.

**Comment by nyngwang** [comment]
> I mean the Neovim installation (wherever that is, e.g., /usr/local/bin/nvim and /usr/local/share/nvim).

But I uninstall it using `brew uninstall neovim`. I have checkout both paths and there are no these folders.

**Comment by gavsim** [comment]
Seeing same behaviour too after installing neovim --HEAD just now.

**Comment by clason** [comment]
I just tested and can't see any errors. Please give a _minimal_ reproducing config and _exact_ steps.

**Comment by nyngwang** [comment]
> I'm closing because it's not reproducible.

Now three people have the same problem. Cool, what a coincident that we were all editing and updating neovim at the same time. Do you try `brew install neovim --HEAD`? @clason

**Comment by clason** [comment]
Yes, of course.

Also, let me be clear: installation issues with `homebrew` are not (and cannot be) supported by Neovim; this is out of our control. And homebrew itself does not offer support for prerelease `HEAD` builds. If you need to run prerelease build, I recommend cloning the repository and building direct from source.

**Comment by nyngwang** [comment]
@clason Thanks for your explanation. But I don't know how to remove all the traces... How can I know where to find them?

**Comment by clason** [comment]
Depends on where homebrew install stuff for you (could be either `/usr/local` or `/opt`, check both).

**Comment by clason** [comment]
OK, I think I see what's going on: https://github.com/neovim/neovim/pull/16631 broke homebrew's patched build scripts (which target the latest release). 

I'm afraid there's nothing we can do about that, and homebrew is also unlikely to patch the patches for HEAD builds (@carlocab).

So for the time being, if you want to use the 0.7.0 prerelease (which, again, I don't recommend to users right now), you _need to_ build straight from the repo (which is easy enough).

**Comment by nyngwang** [comment]
@clason OK, I just built it from the source. Now it works. Sorry for the inconvenience.

**Comment by markuspeloquin** [comment]
Some thing is happening for me in Linux (but only the PANIC). The only two dependencies I use bundled versions of are libvterm and treesitter. I try to use system dependencies for the rest.

**Comment by clason** [comment]
If you don't use homebrew to install, and you don't get the exact same error, that's not the same thing. Please open a new issue (and provide exact and reproducible build steps).

**Comment by markuspeloquin** [comment]
This clearly isn't a homebrew problem. In any case I had to add `-DUSE_BUNDLED_LUA{ROCKS,JIT}=ON` and the build time is probably triple.

**Comment by clason** [comment]
This particular issue _is_ a homebrew problem; please don't highjack unrelated issues.

**Comment by gpanders** [comment]
@markuspeloquin Please open a separate issue as clason suggested. This issue is related to breakage in Homebrew’s build process, if there is similar breakage on Linux systems then we can have a focused discussion in a dedicated issue.

**Comment by gpanders** [comment]
Should be fixed by #16714.

**Comment by carlocab** [comment]
Hi all, sorry for the radio silence here. I'm on holiday. I've opened Homebrew/homebrew-core#91638 as a fix to this (#16714 notwithstanding).

**Comment by nyngwang** [comment]
@carlocab Many thanks~ It's OK for me to install the the `0.6` version for now.

**Comment by carlocab** [comment]
Great, but `HEAD` installs should also be working now after #16714. You just don't get the benefit of bytecode pre-compilation, which my PR will fix.

**Comment by nyngwang** [comment]
@carlocab: Wow! I just got what you meant when I was reading those links. Thank you for saving neovim folks!
